Ex2/Ex2.java

```diff
@@ -0,0 +1,209 @@
+import java.util.HashSet;
+import java.util.Random;
+
+public class Ex2 {
+
+	public static final boolean LEFT = true;
+	public static final boolean RIGHT = false;
+	
+	public static void main(String[] args) {
+		System.out.println("This program actually prints the tree so results can be seen.");
+		System.out.println("Uncomment one of these two lines in \"main\" to check a tree:");
+		System.out.println("//checkWithRandomTree();");
+		System.out.println("//checkWithAssignmentTree();");
+		//checkWithRandomTree();
+		//checkWithAssignmentTree();
+	}
+	
+	/**
+	 * This method prints the ancestors of a given node in the Binary Tree
+	 * @param tree - The Binary Tree Data Structure to work on
+	 * @param targetNode - the node which we would like to print its ancestors
+	 */
+	public static void printAncestors(BinaryTree<?> tree, TreeNode<?> targetNode) {
```

Comment:
I only just realized I can add the generic type <T> in the method signature, as in:
`public static <T> void printAncestors(BinaryTree<T> tree, T key)`
I will try to revise it.